site/app/controllers/admin/GradeableController.php

```diff
@@ -74,4 +99,71 @@ public function process_config_upload() {
         $this->core->redirect($this->core->buildUrl(array('component' => 'admin', 'page' => 'gradeable',
             'action' => 'upload_config')));
     }
+
+    public function rename_config(){
+        $config_file_path = $_POST['curr_config_name'] ?? null;
+        if($config_file_path == null){
+            $this->core->addErrorMessage("Unable to find file");
+
+        } else {
+            $new_name = $_POST['new_config_name'] ?? null;
+            $new_dir = FileUtils::joinPaths(dirname($config_file_path, 1), $new_name);
+            if(rename($config_file_path, $new_dir)){
+                $this->core->addSuccessMessage("Successfully renamed file");
+                $this->core->redirect($this->core->buildUrl(array('component' => 'admin', 'page' => 'gradeable',
+                    'action' => 'upload_config')));
+            } else {
+                $this->core->addErrorMessage("Directory already exist, please choose another name.");
+            }
+        }
+        $this->core->redirect($this->core->buildUrl(array('component' => 'admin', 'page' => 'gradeable',
+            'action' => 'upload_config')));
+    }
+
+    public function delete_config() {
+        $config_path = $_GET['config'] ?? null;
+        if($config_path == null){
+            $this->core->addErrorMessage("Selecting config failed.");
+        } else {
+            if($this->rrmdir($config_path)){
+                $this->core->addSuccessMessage("The config folder has been succesfully deleted");
+            } else {
+                $this->core->addErrorMessage("Deleting config failed.");
+            }
+        }
+        $this->core->redirect($this->core->buildUrl(array('component' => 'admin', 'page' => 'gradeable',
+            'action' => 'upload_config')));
+    }
+
+    private function configUsedBy(){
+        // Returns a list of gradeables that are using this config
+        $config_path = $_GET['config'] ?? null;
+        if($config_path == null){
+            return null;
+        } else {
+            $inuse_config = array();
+            foreach($this->core->getQueries()->getGradeablesIterator() as $gradeable){
+                if($gradeable->getConfigPath() == $config_path){
+                    $inuse_config[] = $gradeable->getId();
+                }
+            }
+        }
+        return $this->core->getOutput()->renderJson($inuse_config);
+    }
+
+    private function rrmdir($dir) {
```

Comment:
Can you add php doc to this function?  Its behavior is not obvious from the name (maybe also lengthen name to `recursive_rmdir`)